### PR TITLE
Properly clear out boost directory when building Docker image

### PR DESCRIPTION
**Pull request by aprokop**
Building boost did not use `SCRATCH_DIR`, and there was no `cd ..`. So when `rm -rf boost*` was executed, we were still inside `/boost`. This commit changes it to approach used when building all other tpls, and blows away `SCRATCH_DIR` at the end.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -110,7 +111,7 @@ RUN BOOST_VERSION=1.67.0 && \
         cxxflags=-w \
         install \
         && \
-    rm -rf boost*
+    rm -rf ${SCRATCH_DIR}
```
Comment (dalg24) [inline docker/Dockerfile:114]:
```suggestion
    cd .. && rm -rf boost*
```

Comment (aprokop) [reply docker/Dockerfile:114]: That's one way. I opted for using `SCRATCH_DIR` to have the same approach as all other blocks in Dockerfile.

File: docker/Dockerfile.pgi [hunk 2]
```diff
@@ -43,7 +43,7 @@ RUN CMAKE_VERSION=3.13.4 && \
     grep ${CMAKE_SCRIPT} ${CMAKE_SHA256} | sha256sum --check && \
     mkdir -p ${CMAKE_DIR} && \
     sh ${CMAKE_SCRIPT} --skip-license --prefix=${CMAKE_DIR} && \
-    rm ${CMAKE_SCRIPT}
+    rm cmake*
```
Comment (masterleinad) [inline docker/Dockerfile.pgi:46]: Do you want `cmake*` as here or `cmake-*` as above?

Comment (dalg24) [reply docker/Dockerfile.pgi:46]: cmake*

**Comment by aprokop** [comment]
retest this please

**Comment by aprokop** [comment]
retest this please

**Comment by aprokop** [comment]
@dalg24 Are you OK with the current status of this PR?

**Comment by dalg24** [comment]
> @dalg24 Are you OK with the current status of this PR?

Still not liking the scratch dir when installing boost.  I am not blocking but not approving either

**Comment by aprokop** [comment]
OK. @masterleinad Can you please approve?

**Comment by aprokop** [comment]
retest this please

**Comment by aprokop** [comment]
retest this please

**Pull request merged**